### PR TITLE
814 feature add api key security to private endpoints on emily openapi spec

### PR DESCRIPTION
**Pull request by AshtonStephens**
## Description

Closes: #814 

Adds api key level security to private Emily API endpoints.

## Changes

- Adds openapi template builder modifiers so the API keys can be added
- Adds the security requirement to the API endpoints 

## Testing Information

- Tested with integration tests and building.

## Checklist:

- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: emily/handler/src/api/handlers/health.rs [hunk 1]
```diff
@@ -15,7 +15,8 @@ use crate::common::error::Error;
         (status = 404, description = "Address not found", body = ErrorResponse),
         (status = 405, description = "Method not allowed", body = ErrorResponse),
         (status = 500, description = "Internal server error", body = ErrorResponse)
-    )
+    ),
+    security(("ApiGatewayKey" = []))
```
Comment (matteojug) [inline emily/handler/src/api/handlers/health.rs:19]: Do we want this to be private?

**Review by matteojug — COMMENTED** [review]
This look good, but it seems to me it's only changing the openapi spec. Is there some aws-magic going on here that actually enforces the api key check?

File: emily/handler/src/api/handlers/health.rs (hunk 1 above)
Comment (AshtonStephens) [reply emily/handler/src/api/handlers/health.rs:19]: I thought so?

**Review by matteojug — APPROVED** [review]

File: emily/handler/src/api/handlers/health.rs (hunk 1 above)
Comment (matteojug) [reply emily/handler/src/api/handlers/health.rs:19]: It seems the kind of endpoint that may be useful for external clients, @aldur WDYT?

Comment (AshtonStephens) [reply emily/handler/src/api/handlers/health.rs:19]: What do you think about us keeping it private for now so we don't need to worry about exposing too much info from this endpoint and then making it public later if we decide we want to?

**Pull request merged**